scripts/west_commands/run_common.py

```diff
@@ -201,6 +252,28 @@ def do_run_common_image(command, user_args, user_runner_args, build_dir=None):
     # parsing, it will show up here, and needs to be filtered out.
     runner_args = [arg for arg in user_runner_args if arg != '--']
 
+    # Check if there are any commands that should only be ran once per board
+    # and if so, remove them for all but the first iteration of the flash
+    # runner per unique board name.
+    try:
+        if len(used_cmds) > 0 and len(runner_args) > 0:
+            for a in runner_args:
+                i = 0
+                while i < len(used_cmds):
+                    try:
+                        if used_cmds[i][a] and type(used_cmds[i][a].index(board)):
+                            if used_cmds[i]['Ran'] is False:
+                                used_cmds[i]['Ran'] = True
+                            else:
+                                runner_args.pop(runner_args.index(a))
+                    except ValueError:
+                        pass
+
+                    i = i + 1
```

Comment:
It seems like making a dataclass for each used_cmds element, and then iterating over it, would be better. I think the current thing is not super readable. E.g.

```py
from dataclasses import dataclass

@dataclass
class UsedCommand:
    command: str
    ran: bool = False

used_cmds = [UsedCommand(f'hello {i}', i % 2 == 0) for i in range(2)]

for i, used_cmd in enumerate(used_cmds):
    print(f'{i=}')
    print(f'\t{used_cmd=}')
    used_cmd.ran = True
    print(f'\t{used_cmd=}')
```

prints:

```

i=0
	used_cmd=UsedCommand(command='hello 0', ran=True)
	used_cmd=UsedCommand(command='hello 0', ran=True)
i=1
	used_cmd=UsedCommand(command='hello 1', ran=False)
	used_cmd=UsedCommand(command='hello 1', ran=True)
```

although it doesn't seem like you really need the index `i` and could just do `for used_cmd in used_cmds:`; right?